### PR TITLE
Allow await in declarations in C# Interactive

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @tmat, @kevinh-ms

File: src/Compilers/CSharp/Portable/Binder/Binder_Await.cs [hunk 1]
```diff
@@ -116,45 +116,49 @@ private bool ContextForbidsAwait
...
+                            return false;
+                        }
+                        break;
+                    case SymbolKind.Method:
+                        var method = (MethodSymbol)containingMemberOrLambda;
+                        if (method.IsAsync)
+                        {
+                            return false;
+                        }
+                        if (!method.IsAsync)
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/Binder_Await.cs]: Redundant?

**Comment by amcasey** [comment]
:+1:

File: src/Compilers/CSharp/Portable/Binder/Binder_Await.cs (hunk 1 above)
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/Binder_Await.cs]: Removed.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs [hunk 2]
```diff
@@ -3400,27 +3400,44 @@ class Program
             CompileAndVerify(comp.WithOptions(TestOptions.ReleaseExe));
         }
 
+        /// <summary>
+        /// await should be disallowed in top-level
+        /// script statements since the script main
+        /// method should complete synchronously.
+        /// </summary>
```
Comment (KevinH-MS) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs]: I'd expect this to work...as well as:

```> await System.Threading.Tasks.Task.FromResult(1)```
```1```

Comment (cston) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncTests.cs]: Logged #4495 to change the Script entry point to block until the async initializer completes.

**Comment by KevinH-MS** [comment]
:+1:

**Comment by cston** [comment]
@dotnet-bot retest this please

**Pull request merged**